File: articles/individuation-and-the-development-of-the-depth-self/go/individuation_score.go
package main

import (
	"fmt"
	"math"
)

func oneSidedness(
	egoCoherence,
	shadowAcknowledgment,
	symbolicRelation,
	complexActivation,
	personaIdentification,
	reflectiveCapacity float64,
) float64 {
	a := egoCoherence - shadowAcknowledgment
	b := symbolicRelation - complexActivation
	c := personaIdentification - reflectiveCapacity
	return a*a + b*b + c*c
}

func depthSelfIntegration(
	egoCoherence,
	shadowAcknowledgment,
	symbolicRelation,
	reflectiveCapacity,
	bodyAwareness,
	ethicalAccountability,
	relationalLife,
	complexActivation,
	oneSidednessValue float64,
) float64 {
	return 0.54*egoCoherence +
		0.62*shadowAcknowledgment +
		0.66*symbolicRelation +
		0.52*reflectiveCapacity +
		0.34*bodyAwareness +
		0.36*ethicalAccountability +
		0.30*relationalLife -
		0.42*math.Abs(complexActivation) -
		0.28*oneSidednessValue
}

func main() {
	w := oneSidedness(0.84, 0.86, 0.92, 0.28, 0.30, 0.90)
	score := depthSelfIntegration(0.84, 0.86, 0.92, 0.90, 0.76, 0.82, 0.82, 0.28, w)

	fmt.Printf("Synthetic one-sidedness: %.3f\n", w)
	fmt.Printf("Synthetic depth-self integration score: %.3f\n", score)
}

File: articles/individuation-and-the-development-of-the-depth-self/rust/individuation_score.rs
fn one_sidedness(
    ego_coherence: f64,
    shadow_acknowledgment: f64,
    symbolic_relation: f64,
    complex_activation: f64,
    persona_identification: f64,
    reflective_capacity: f64,
) -> f64 {
    let a = ego_coherence - shadow_acknowledgment;
    let b = symbolic_relation - complex_activation;
    let c = persona_identification - reflective_capacity;
    a * a + b * b + c * c
}

fn depth_self_integration(
    ego_coherence: f64,
    shadow_acknowledgment: f64,
    symbolic_relation: f64,
    reflective_capacity: f64,
    body_awareness: f64,
    ethical_accountability: f64,
    relational_life: f64,
    complex_activation: f64,
    one_sidedness_value: f64,
) -> f64 {
    0.54 * ego_coherence
        + 0.62 * shadow_acknowledgment
        + 0.66 * symbolic_relation
        + 0.52 * reflective_capacity
        + 0.34 * body_awareness
        + 0.36 * ethical_accountability
        + 0.30 * relational_life
        - 0.42 * complex_activation.abs()
        - 0.28 * one_sidedness_value
}

fn main() {
    let w = one_sidedness(0.84, 0.86, 0.92, 0.28, 0.30, 0.90);
    let score = depth_self_integration(0.84, 0.86, 0.92, 0.90, 0.76, 0.82, 0.82, 0.28, w);

    println!("Synthetic one-sidedness: {:.3}", w);
    println!("Synthetic depth-self integration score: {:.3}", score);
}
